### ISSUE TITLE
[a11y]: OverflowMenu [default] - render custom icon bug 

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#15259]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.40.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.39.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Description
"After selection , Screen reader not announced the selected value of the  Overflow Menu.
Steps to reproduce :
1. Turn on  Mouse and enable screen reader , and reach Targeted  Test component .
2. Select any option from the  Overflow menu .
3. Now  navigate over the same  "Overflow  Menu " which we done selection , and observe the screen reader announcement -
Screen reader not announcing the selected value."

### Recommended Fix
Make screen reader announced the selected value of the  Overflow menu

If the screen reader fails to identify the option selected in the drop-down list, it impacts user interactivity.

### WCAG 2.1 Violation
4.1.2 Name, Role, Value
